src/storage/s3.rs

```diff
@@ -84,9 +92,15 @@ impl<'a> S3Backend<'a> {
             }
             attempts += 1;
 
-            match self.runtime.block_on(futures.map(drop).collect()) {
+            let result: Result<Vec<PutObjectOutput>, RusotoError<PutObjectError>> = self
+                .runtime
+                .block_on(future::join_all(futures))
```

Comment:
Given the potential batch sizes this should use `FuturesUnordered` still.

Reply:
And using `.try_for_each(drop)` or something similar is still useful to discard the unused success results instead of allocating memory to store them.